src/main/java/twilightforest/block/ColorHandler.java

```diff
@@ -263,9 +265,18 @@ public static void init() {
 
 		ItemColors itemColors = Minecraft.getMinecraft().getItemColors();
 		// Atomic: This is one place where getStateFromMeta is still commonly used
-		itemColors.registerItemColorHandler((stack, tintIndex) -> blockColors.colorMultiplier(((ItemBlock)stack.getItem()).getBlock().getStateFromMeta(stack.getMetadata()), null, null, tintIndex), TFBlocks.auroraBlock, TFBlocks.auroraPillar, TFBlocks.auroraSlab, TFBlocks.auroraDoubleSlab, TFBlocks.darkleaves, TFBlocks.giantLeaves, TFBlocks.fireJet, TFBlocks.magicLeaves, TFBlocks.leaves, TFBlocks.leaves3, TFBlocks.plant, TFBlocks.castleMagic, TFBlocks.castleDoor, TFBlocks.castleDoorVanished);
+		itemColors.registerItemColorHandler((stack, tintIndex) -> colorMultiplierFromItem(blockColors, stack, tintIndex), TFBlocks.auroraBlock, TFBlocks.auroraPillar, TFBlocks.auroraSlab, TFBlocks.auroraDoubleSlab, TFBlocks.darkleaves, TFBlocks.giantLeaves, TFBlocks.fireJet, TFBlocks.magicLeaves, TFBlocks.leaves, TFBlocks.leaves3, TFBlocks.plantStandard, TFBlocks.plantHanging, TFBlocks.plantDark, TFBlocks.plantCave, TFBlocks.castleMagic, TFBlocks.castleDoor, TFBlocks.castleDoorVanished);
 		// Honestly I'd say it makes sense in this context. -Drullkus
 	}
+	
+	private static int colorMultiplierFromItem( BlockColors blockColors, ItemStack stack, int tintIndex ) {
+		// NOTE: Handling possible forge bug. 
+		Item item = stack.getItem();
+		if( !(item instanceof ItemBlock) )
+			return -1;
+		
+		return blockColors.colorMultiplier(((ItemBlock)stack.getItem()).getBlock().getStateFromMeta(stack.getMetadata()), null, null, tintIndex);
+	}
```

Comment:
? is there an actual reproducible issue here? If so, report it to forge and put a comment to the forge issue here.

Reply:
Isn't this due to the Aurora double slab not having an itemblock, but this was already fixed in the 1.12 branch?

Reply:
It's reproducible in MC 1.11.2 Version, when picking up a lot of items at once. A random crash occurrs with a ClassCastException.

It's not known whether it is a forge bug or related to any coinstalled mod, like Optifine or s. e. . Sometimes invalid ItemStacks are passed containing items, having types which are not listed up in the argument list of a registerItemColorHandler call.